### ISSUE TITLE
Details Block: Inner blocks cannot be focused using only the keyboard

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#70047]
### Description

Reported at https://github.com/WordPress/gutenberg/issues/65603#issuecomment-2844575410

Previously, the Details block automatically opened and closed the contents inside the block when the block was selected. However, this behavior was removed by #67766 after users reported issues with dragging and dropping blocks.

However, this introduced another accessibility issue. Keyboard users cannot access the inner blocks because the contents inside the block can only be opened and closed by clicking the summary element with the mouse pointer.

### Step-by-step reproduction instructions

1. Add a new post.
2. Insert details block.
3. Notice how focus should land in the summary title field.
4. Try to move to the hidden paragraph.
5. Impossible because the block is collapsed.
6. Try to expand the block with the keyboard.
7. Impossible because there is no exposed expander/trigger due to the overriding of semantics for screen readers.
8. Open the list view.
9. Expand the details block in the tree.
10. Try to select the paragraph block.
11. Block is announced as selected but focus doesn't move since it is not exposed in the canvas.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by t-hamano** [comment]
Another issue: I also discovered that when the Details block's content is not expanded, it is not possible to move the block with the up and down keys.

Example HTML:

```html
<!-- wp:paragraph -->
<p>First Paragraph Block</p>
<!-- /wp:paragraph -->

<!-- wp:details -->
<details class="wp-block-details">
<summary>Details Block Summary</summary>
<!-- wp:paragraph {"placeholder":"Type / to add a hidden block"} -->
<p>Details Block Content</p>
<!-- /wp:paragraph -->
<!-- wp:paragraph -->
<p>Details Block Content</p>
<!-- /wp:paragraph -->
</details>
<!-- /wp:details -->

<!-- wp:paragraph -->
<p>Second Paragraph Block</p>
<!-- /wp:paragraph -->
```

Step-by-step reproduction instructions:

- Insert a first Paragraph block, a Details block with contents, and a second Paragraph block.
- Focus the post title.
- Press the `down` key. The First Paragraph block is focused.
- Press the `down` key. The Details Paragraph block is focused.
- Press the `down` key. The `summary` element is focused.
- Press the `down` key repeatedly. The second Paragraph block can't get focused.

**Comment by armandovias** [comment]
Thank you for opening this issue.

**Comment by armandovias** [comment]
This bug is also occurring with #70140 .
